std/meta/list.d

```diff
@@ -0,0 +1,1038 @@
+// Written in the D programming language.
+
+/**
+ * Templates with which to manipulate
+ * $(LINK2 ../template.html#TemplateArgumentList, $(I TemplateArgumentList))s.
+ * Such lists are known as type lists when they only contain types.
+ *
+ * Some operations on template argument lists are built in to the language,
+ * such as $(D Args[$(I n)]) which gets the $(I n)th element from the
+ * _list. $(D Args[$(I lwr) .. $(I upr)]) returns a new
+ * _list that is a slice of the old one.
+ *
+ * Several templates in this module use or operate on enum templates that
+ * take a single argument and evaluate to a boolean constant. Such templates
+ * are referred to as $(I template predicates).
+ * 
+ * Some symbol names defined in this module may conflict with e.g.
+ * $(D std.algorithm) symbol names. The recommended practice is
+ * therefore to import this module with a $(D meta) alias:
+ * ---
+ * import $(B $(I meta)) = std.meta.list;
+ * 
+ * alias Types = meta.List!(byte, int, char[]);
+ * static assert(meta.indexOf!(int, Types) == 1);
+ * ---
+ *
+ * References:
+ *  Based on ideas in Table 3.1 from
+ *  $(LINK2 http://amazon.com/exec/obidos/ASIN/0201704315/ref=ase_classicempire/102-2957199-2585768,
+ *      Modern C++ Design),
+ *   Andrei Alexandrescu (Addison-Wesley Professional, 2001)
+ * Macros:
+ *  WIKI = Phobos/StdMeta
+ *
+ * Copyright: Copyright Digital Mars 2005 - 2009.
+ * License:   $(WEB www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+ * Authors:
+ *     $(WEB digitalmars.com, Walter Bright),
+ *     $(WEB klickverbot.at, David Nadlinger)
+ * Source:    $(PHOBOSSRC std/meta/_list.d)
+ */
+/*          Copyright Digital Mars 2005 - 2009.
+ * Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ */
+module std.meta.list;
+
+/**
+ * Aliases the given compile-time list of template arguments.
+ */
+alias List(Args...) = Args;
+
+///
+unittest
+{
+    alias TL = List!(int, double);
+
+    int foo(TL td)  // same as int foo(int, double);
+    {
+        return td[0] + cast(int)td[1];
```

Comment:
Side note: This is a good example for `std.traits.ParameterTypeTuple` (which has no example on how to access the parameters).